javascript.js

```diff
@@ -12,3 +12,157 @@ hamburger.addEventListener('click', display);
+const nextPrevious = document.createElement('div');
+const previous = document.createElement('img');
+const next = document.createElement('img');
+
+section.className = 'default';
+xIcon.className = 'close-icon';
+heading.className = 'popup-heading';
+list.className = 'popup-list';
+listItem1.classList.add('popup-list-item', 'popup-list-item1');
+listItem2.classList.add('popup-list-item', 'popup-list-item2');
+listItem3.classList.add('popup-list-item', 'popup-list-item3');
+slideshow.className = 'slideshow';
+arrowLeft.className = 'arrow';
+arrowRight.className = 'arrow';
+slides.className = 'slides';
+image1.className = 'slide';
+image2.className = 'slide';
+image3.className = 'slide';
+image4.className = 'slide';
+flexContainer.className = 'center';
+paragraph1.className = 'about-project';
+paragraph2.className = 'about-project';
+buttons.className = 'popup-buttons';
+button1.className = 'live-source-buttons';
+span1.className = 'live-source-span';
+liveImage.className = 'live-image';
+button2.className = 'live-source-buttons';
+span2.className = 'live-source-span';
+sourceImage.className = 'live-image';
+nextPrevious.className = 'next-previous';
+previous.className = 'previous';
+next.className = 'next';
+
+xIcon.src = project.closePopup;
+heading.textContent = project.name;
+listItem1.textContent = project.item1;
+listItem2.textContent = project.item2;
+listItem3.textContent = project.item3;
+arrowLeft.src = project.leftArrow;
+arrowRight.src = project.rightArrow;
+image1.src = project.imageURL;
+image2.src = project.imageURL;
+image3.src = project.imageURL;
+image4.src = project.imageURL;
+paragraph1.textContent = project.description;
+paragraph2.textContent = project.description2;
+link1.href = project.liveVersionLink;
+span1.textContent = project.live;
+liveImage.src = project.liveImage;
+link2.href = project.sourceLink;
+span2.textContent = project.source;
+sourceImage.src = project.sourceImage;
+previous.src = project.previous;
+next.src = project.next;
+
+document.body.appendChild(section);
+section.appendChild(xIcon);
+section.appendChild(heading);
+section.appendChild(list);
+section.appendChild(slideshow);
+list.appendChild(listItem1);
+list.appendChild(listItem2);
+list.appendChild(listItem3);
+slideshow.appendChild(arrowLeft);
+slideshow.appendChild(arrowRight);
+section.appendChild(slides);
+slides.appendChild(image1);
+slides.appendChild(image2);
+slides.appendChild(image3);
+slides.appendChild(image4);
+section.appendChild(flexContainer);
+flexContainer.appendChild(paragraph1);
+flexContainer.appendChild(paragraph2);
+section.appendChild(buttons);
+buttons.appendChild(button1);
+button1.appendChild(link1);
+link1.appendChild(span1);
+link1.appendChild(liveImage);
+buttons.appendChild(button2);
+button2.appendChild(link2);
+link2.appendChild(span2);
+link2.appendChild(sourceImage);
+section.appendChild(nextPrevious);
+nextPrevious.appendChild(previous);
+nextPrevious.appendChild(next);
```

Comment:
In order to avoid over 112 lines of code, we could seek another approach when we create the popup 🤓 Check out the following [link](https://wesbos.com/template-strings-html) so that you can find more concrete guidance. A quick example:

    const div = document.createElement('div');
    div.innerHTML = '<img src="./images/buttons/Icon.png" class="cancelBttn"></img>'
    + '<ul class="mobile-menu-ul">'
    + '<li class="mobile-menu-li"><a href="#section-1">Porfolio</a></li>'
    + '<li class="mobile-menu-li"><a href="#section-2">About</a></li>'
    + '<li class="mobile-menu-li"><a href="#section-3">Contact</a></li>'
    + '</ul>';
    document.body.appendChild(div);

Please make the necessary changes so that the code follow JS best practices 😁